src/main/java/dk/bec/polonez/reservationsystem/service/ReservationService.java

```diff
@@ -25,26 +28,114 @@
 
     private final OfferRepository offerRepository;
 
+    private final AuthService authService;
 
-    public ReservationService(ReservationRepository reservationRepository, UserRepository userRepository, OfferRepository offerRepository) {
+    private final ModelMapper modelMapper;
+
+    //TODO: check if reservation collides with other ones on that offer during addition
+
+
+    public ReservationService(ReservationRepository reservationRepository, UserRepository userRepository, OfferRepository offerRepository, AuthService authService) {
         this.reservationRepository = reservationRepository;
         this.userRepository = userRepository;
         this.offerRepository = offerRepository;
+        this.authService = authService;
+        this.modelMapper = new ModelMapper();
+
+        PropertyMap<Reservation, ResponseReservationDto> responseReservationMap = new PropertyMap<Reservation, ResponseReservationDto>() {
+            @Override
+            protected void configure() {
+                map().setOfferId(source.getOffer().getId());
+                map().setUserId(source.getUser().getId());
+            }
+        };
+
+        modelMapper.addMappings(responseReservationMap);
+    }
+
+    public ArrayList<ResponseReservationDto> getAll(Boolean sortedByDate) throws ResponseStatusException{
```

Comment:
**advice** It's better approach to use interface ```List``` instead of implementation type ```ArrayList``` in the method signature.